controller/cache/info.go

```diff
@@ -38,13 +38,24 @@ func populateNodeInfo(un *unstructured.Unstructured, res *ResourceInfo, customLa
 		}
 	}
 
+	// always apply resources update by default
+	res.applyResourcesUpdate = true
+
 	for k, v := range un.GetAnnotations() {
 		if strings.HasPrefix(k, common.AnnotationKeyLinkPrefix) {
 			if res.NetworkingInfo == nil {
 				res.NetworkingInfo = &v1alpha1.ResourceNetworkingInfo{}
 			}
 			res.NetworkingInfo.ExternalURLs = append(res.NetworkingInfo.ExternalURLs, v)
 		}
+
+		if k == AnnotationApplyResourcesUpdate {
+			value, err := strconv.ParseBool(v)
+			if err != nil {
+				value = false
+			}
+			res.applyResourcesUpdate = value
+		}
```

Comment:
The code can be simplified by just checking this at the end of `shouldHashManifest` method.
If the shouldHashManifest return true because the annotations is present for an untracked resource, then it will take the same codepath as current behavior for tracked resources. 

No need to update the ResourceInfo, store booleans or annotations.

Reply:
@agaudreault updated with your suggestion. 

One additional change in `skipResourceUpdate`: since we are not going to generate hash and therefore the logic of `isSameManifest` need to change. So PTAL.

Reply:
We simply cannot generate a hash by default or skip the resources updates for all the resources. When the annotation is not defined, the hash should not be generated. If you generate a hash by default for all resource updates, the controller will use a lot more CPU generating hashes than what you will save skipping reconcile.

I also think we should not go in a direction where non-argocd resources must have argocd annotations. 

That is why I think the best approach is:
- Generate the hash only if annotation is present and true.
- (already done) Skip the resource updates if the hashes exist and the hashes match OR if the resource does not belong to any aplication